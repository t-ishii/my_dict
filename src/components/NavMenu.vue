<template>
  <el-menu :default-active="activeIndex" :router="true" mode="horizontal" @select="handleSelect">
    <li role="menuitem" class="el-menu-item"><el-input 
      placeholder="Keyword"
      suffix-icon="el-icon-search"></el-input></li>
    <el-menu-item index="/">
      <i class="el-icon-menu"></i>
    </el-menu-item>
    <el-menu-item index="/new">
      <i class="el-icon-circle-plus-outline"></i>
    </el-menu-item>
    <el-menu-item index="/about">
      <i class="el-icon-question"></i>
    </el-menu-item>
    <el-submenu index="setting">
      <template slot="title"><i class="el-icon-setting"></i></template>
      <el-menu-item index="/login">logout</el-menu-item>
    </el-submenu>
  </el-menu>
</template>
<script>
import firebase from 'firebase'
import 'firebase/auth'

export default {
  name: 'NavMenu',
  data () {
    return {
      activeIndex: '/'
    }
  },
  methods: {
    handleSelect (key) {
      if (key === '/login') {
        firebase.auth().signOut()
      }
    }
  }
}
</script>
