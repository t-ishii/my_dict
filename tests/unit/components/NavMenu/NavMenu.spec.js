import { shallowMount, createLocalVue } from '@vue/test-utils'
import NavMenu from '@/components/NavMenu.vue'
import Vuex from 'vuex'
import VueRouter from 'vue-router'

import ElementUI from 'element-ui'
import locale from 'element-ui/lib/locale/lang/ja'

const localVue = createLocalVue()
localVue.use(ElementUI, { locale })
localVue.use(Vuex)
localVue.use(VueRouter)

const store = new Vuex.Store({
  modules: {
    user: {
      state: {
        isLogin: false
      },
      mutations: {
        logout(state) {
          state.isLogin = false
        },
        login(state) {
          state.isLogin = true
        }
      }
    }
  }
})

const router = new VueRouter({
  routes: [
    {
      path: '/home',
      name: 'home',
      component: NavMenu
    }
  ]
})

describe('NavMenu.vue', () => {

  // ログインしていないときに、aboutアイコンだけ表示されているか
  it('Whether only the about icon is displayed when not logged in.', () => {
    const wrapper = shallowMount(NavMenu, {
      store,
      localVue
    })
    expect(wrapper.find('#nav-home').exists()).toBeFalsy()
    expect(wrapper.find('#nav-new').exists()).toBeFalsy()
    expect(wrapper.find('#nav-about').exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  // ログイン時にアイコンが表示されるかどうか
  it('Whether icons is displayed when logged in.', () => {
    store.state.user.isLogin = true
    const wrapper = shallowMount(NavMenu, {
      store,
      localVue
    })
    expect(wrapper.find('#nav-home').exists()).toBeTruthy()
    expect(wrapper.find('#nav-new').exists()).toBeTruthy()
    expect(wrapper.find('#nav-about').exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

})
