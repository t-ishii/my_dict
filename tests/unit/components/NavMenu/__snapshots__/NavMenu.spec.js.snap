// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu.vue Whether icons is displayed when logged in. 1`] = `
<div class="header-wrapper">
  <elmenu-stub mode="horizontal" defaultactive="/" router="true" menutrigger="hover" collapsetransition="true">
    <elmenuitem-stub index="/home" id="nav-home"><i class="el-icon-menu"></i></elmenuitem-stub>
    <elmenuitem-stub index="/new" id="nav-new"><i class="el-icon-edit"></i></elmenuitem-stub>
    <elmenuitem-stub index="/about" id="nav-about"><i class="el-icon-question"></i></elmenuitem-stub>
    <elsubmenu-stub index="setting" showtimeout="300" hidetimeout="300"><template></template>
      <elmenuitem-stub index="/login">logout</elmenuitem-stub>
    </elsubmenu-stub>
  </elmenu-stub>
</div>
`;

exports[`NavMenu.vue Whether only the about icon is displayed when not logged in. 1`] = `
<div class="header-wrapper">
  <elmenu-stub mode="horizontal" defaultactive="/" router="true" menutrigger="hover" collapsetransition="true">
    <!---->
    <elmenuitem-stub index="/about" id="nav-about"><i class="el-icon-question"></i></elmenuitem-stub>
    <!---->
  </elmenu-stub>
</div>
`;
